driver/web/rest/adminapis.go

```diff
@@ -1387,3 +1387,460 @@ func (h AdminApisHandler) GetContentItemsCategories(claims *tokenauth.Claims, w
 	w.WriteHeader(http.StatusOK)
 	w.Write(data)
 }
+
+// CreateDataContentItem Creates a new data content type item
+// @Description Creates a new data content type item
+// @Tags Admin
+// @ID AdminCreateDataContentItem
+// @Accept json
+// @Success 200
+// @Security AdminUserAuth
+// @Router /admin/data [post]
+func (h AdminApisHandler) CreateDataContentItem(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error on marshal create a data content item - %s\n", err.Error())
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	var item model.DataContentItem
+	err = json.Unmarshal(data, &item)
```

Comment:
Let's use `json.NewDecoder(r.Body).Decode(&item)` wherever it's needed instead of reading and unmarshalling separately.